vaadin-card-flow-parent/vaadin-card-flow/src/main/java/com/vaadin/flow/component/card/Card.java

```diff
@@ -32,6 +49,358 @@
+        if (index < 0) {
+            throw new IllegalArgumentException(
+                    "Cannot add a component with a negative index");
+        }
+        if (contentRoot == null) {
+            initContentRoot();
+        }
+        contentRoot.insertChild(index, component.getElement());
+    }
+
+    /**
+     * Sets the ARIA role attribute on the card.
+     *
+     * @param role
+     *            the ARIA role, or {@code null} to clear
+     */
+    public void setAriaRole(String role) {
+        if (role == null) {
+            getElement().removeAttribute("role");
+        } else {
+            getElement().setAttribute("role", role);
+        }
+    }
+
+    /**
+     * Gets the ARIA role attribute of the card.
+     *
+     * @return an optional ARIA role of the card if no ARIA role has been set
+     */
+    public Optional<String> getAriaRole() {
+        return Optional.ofNullable(getElement().getAttribute("role"));
+    }
+
+    @Override
+    protected void onAttach(AttachEvent attachEvent) {
+        super.onAttach(attachEvent);
+        checkFeatureFlag();
+    }
+
+    /**
+     * Gets the feature flags for the current UI.
+     * <p>
+     * Not private in order to support mocking
+     *
+     * @return the current set of feature flags
+     */
+    FeatureFlags getFeatureFlags() {
+        return FeatureFlags
+                .get(UI.getCurrent().getSession().getService().getContext());
+    }
+
+    /**
+     * Only for test use.
+     */
+    void setFeatureFlagEnabled() {
+        featureFlagEnabled = true;
+    }
+
+    /**
+     * Checks whether the Card component feature flag is active. Succeeds if the
+     * flag is enabled, and throws otherwise.
+     *
+     * @throws ExperimentalFeatureException
+     *             when the {@link FeatureFlags#CARD_COMPONENT} feature is not
+     *             enabled
+     */
+    private void checkFeatureFlag() {
+        boolean enabled = featureFlagEnabled
+                || getFeatureFlags().isEnabled(FeatureFlags.CARD_COMPONENT);
+        if (!enabled) {
+            throw new ExperimentalFeatureException();
+        }
+    }
+
+    private void setSlotContent(Component slotContent, String slotName) {
+        if (slotContent == null) {
+            SlotUtils.getElementsInSlot(this, slotName).findFirst()
+                    .ifPresent(Element::removeAllChildren);
+            SlotUtils.clearSlot(this, slotName);
+            return;
+        }
+        var slotElement = new AtomicReference<Element>();
+        SlotUtils.getElementsInSlot(this, slotName).findFirst().ifPresentOrElse(
+                slotElement::set, () -> slotElement.set(initSlot(slotName)));
+        slotElement.get().removeAllChildren();
+        slotElement.get().appendChild(slotContent.getElement());
+    }
+
+    private Element initSlot(String slotName) {
+        var div = new Element("div");
+        SlotUtils.setSlot(this, slotName, div);
```

Comment:
Is there a particular reason for adding an extra `<div>` wrapper for each slot? Could the slot attribute be assigned directly to the component element instead?

Reply:
Not really. One small benefit is to enable the use of Text component directly. But removing it also makes everything much simpler. Therefore, I removed the use of the wrappers.

Reply:
Can you think of some potential issue of putting all of the main content into a wrapper div? I couldn't see anything from a quick glance, but it seems worth double-checking.

Reply:
One concern is that wrappers might complicate styling in the future because the `::slotted` selector only works with direct children. Also, if a slot uses `flexbox` or other display models, its children won't automatically inherit it unless the wrapper is given `display: contents`...

Reply:
Is there a practical reason to wrap content at the moment though?

Reply:
How do you mean? I'm assuming we're wrapping the content because it makes implementing `HasComponents` a lot easier for stuff like `addComponentAtIndex`.

Reply:
If we want to avoid a wrapper then `addComponentAtIndex` seems like the only challenge. Everything else could maybe use the default implementation. For `removeAll` it's kind of debatable whether that should remove everything or just children from the content slot.

Reply:
Updated the PR to add `display: contents` to the content wrapper.